messaging/internal/service/webhook_test.go

```diff
@@ -0,0 +1,87 @@
+// +build integration,external
+
+package service
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	"github.com/crusttech/crust/internal/auth"
+	"github.com/crusttech/crust/internal/config"
+	"github.com/crusttech/crust/internal/http"
+	"github.com/crusttech/crust/internal/test"
+	"github.com/crusttech/crust/messaging/internal/repository"
+	"github.com/crusttech/crust/messaging/types"
+	systemService "github.com/crusttech/crust/system/service"
+	systemTypes "github.com/crusttech/crust/system/types"
+)
+
+func TestOutgoingWebhook(t *testing.T) {
+	var user = &systemTypes.User{ID: 1}
+	var channel = &types.Channel{ID: 1}
+
+	ctx := context.WithValue(context.Background(), "testing", true)
+	ctx = auth.SetIdentityToContext(ctx, user)
+
+	// set up user
+	{
+		svc := systemService.User(ctx)
+		_, err := svc.Create(user)
+		test.Assert(t, err == nil, "Error when creating user: %+v", err)
+	}
+
+	// set up channel
+	{
+		rpo := repository.Channel(ctx, repository.DB(ctx))
+		_, err := rpo.Create(channel)
+		test.Assert(t, err == nil, "Error when creating channel: %+v", err)
+	}
+
+	client, err := http.New(&config.HTTPClient{
+		Timeout: 10,
+	})
+	test.Assert(t, err == nil, "Error creating HTTP client: %+v", err)
+
+	svc := Webhook(ctx, client)
+
+	{
+		/* create outgoing webhook */
+		webhook, err := svc.Create(types.OutgoingWebhook, channel.ID, types.WebhookRequest{
+			Username:        "test-webhook",
+			OutgoingTrigger: "fortune",
+			OutgoingURL:     "https://api.scene-si.org/fortune.php",
```

Comment:
this should be tested with internal fake http server.

Reply:
Agreed, test is marked as external and isn't part of the CI tests. Going to open up an issue for this.

Reply:
got this today (while running tests/drone locally):

```
[tests:191] HTTP <<< Error: Get https://api.scene-si.org/fortune.php: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)
```